cpg-core/src/main/java/de/fraunhofer/aisec/cpg/frontends/Language.kt

```diff
@@ -58,12 +65,84 @@ abstract class Language<T : LanguageFrontend> : Node() {
     open val primitiveTypes: Set<String>
         get() = setOf("byte", "short", "int", "long", "float", "double", "boolean", "char")
 
+    /** The access modifiers of this programming language */
+    open val accessModifiers: Set<String>
+        get() = setOf("public", "protected", "private")
+
     /** Creates a new [LanguageFrontend] object to parse the language. */
     abstract fun newFrontend(
         config: TranslationConfiguration,
         scopeManager: ScopeManager = ScopeManager()
     ): T
 
+    open fun getTypeOf(typeString: String, modifier: ObjectType.Modifier): Type? {
```

Comment:
the naming should be something related to "simple" types I guess.

I also wonder, whether it would make more sense to move this to a `LanguageFrontend`? I would rather keep "logic" out of the language class.

Alternatively, we could provide a `Map` that maps between a name/string and a type? That would suit the configuration style of the language class better I would say.

Reply:
I would not want to move it to the `LanguageFrontend` because we don't know when this method or the `TypeParser` will be used. I.e., if it is in a Pass, the frontend won't be available. But the idea with the `Map` sounds good (and I also thought of that already)